packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/src/Shared/StringExtensions.cs

```diff
@@ -244,5 +244,50 @@ public static string ToApiVersionMemberName(this string version)
 
             return CultureInfo.InvariantCulture.TextInfo.ToTitleCase(sb.ToString());
         }
+
+        /// <summary>
+        /// Checks if two namespaces share the same last segment
+        /// </summary>
+        /// <param name="left">the first namespace</param>
+        /// <param name="right">the second namespace</param>
+        /// <returns></returns>
+        public static bool IsLastNamespaceSegmentTheSame(string left, string right)
```

Comment:
I introduce this method to verify if the namespace has been changed in one client